pkg/util/encoding/encoding.go

```diff
@@ -44,11 +44,12 @@ const (
 	// it's used in keys used as span boundaries for index scans.
 	encodedNotNull = 0x01
 
-	floatNaN     = encodedNotNull + 1
-	floatNeg     = floatNaN + 1
-	floatZero    = floatNeg + 1
-	floatPos     = floatZero + 1
-	floatNaNDesc = floatPos + 1 // NaN encoded descendingly
+	floatNaNDesc    = encodedNotNull + 1 // NaN encoded descendingly
```

Comment:
This changes the meaning for values that are already on disk. We essentially cannot change this package. Or rather, if you do change this package, it must meet some properties:

1) Data encoded in the old format must be understandable by any future reader.
2) Data encoded in the new format must be understandable by the previous version, or the new data format will not be written until all nodes have been upgraded to the current version.

This PR violates both of those properties.

Reply:
(I think this is a very tricky issue, and I suggest you consider not working on it due to its annoyance.)